sdk/go/client.go

```diff
@@ -1,93 +1,112 @@
 package ehsm
 
 import (
-    tls "crypto/tls"
-    "encoding/json"
-    "fmt"
-    http "net/http"
-    "sort"
-    "os"
-    "bytes"
-    "sync"
+	"bytes"
+	hmac "crypto/hmac"
+	sha256 "crypto/sha256"
+	tls "crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	http "net/http"
+	"os"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
 
-    "github.com/iancoleman/orderedmap"
+	"github.com/iancoleman/orderedmap"
 )
 
 type Client struct {
-    modifyLock              sync.RWMutex
-    addr                    string
-    apikey                  string
-    appid                   string
-    action                  string
-    resquest                *orderedmap.OrderedMap
+	modifyLock sync.RWMutex
+	addr       string
+	apikey     string
+	appid      string
+	action     string
+	resquest   *orderedmap.OrderedMap
```

Comment:
why define the request in this structure? 

can we move it as a input params for ehsmHttpAction?

Reply:
done